sklearn/dummy.py

```diff
@@ -141,7 +142,10 @@ def fit(self, X, y, sample_weight=None):
 
         self.n_outputs_ = y.shape[1]
 
-        check_consistent_length(X, y, sample_weight)
+        check_consistent_length(X, y)
```

Comment:
I think we might still want it here, not strong opinion, though...

Reply:
even though it's checked as part of the added validation?

Reply:
+1 to avoid redundant checks (even if it doesn't cost much). `_check_sample_weight` should yield better error messages